packages/tree-view/src/ux-tree-view/ux-tree-view.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  customElement, bindable, useView, PLATFORM, processContent, ViewCompiler, ViewResources, BehaviorInstruction,
+  inject, Optional, Container, ViewFactory
+} from 'aurelia-framework';
+import { INode } from './i-node';
+import { UxComponent, StyleEngine } from '@aurelia-ux/core';
+import { UxTreeViewTheme } from './ux-tree-view-theme';
+import { UxDefaultTreeViewConfiguration } from '../ux-default-tree-view-configuration';
+
+let id = 0;
+const templateLookup: Record<string, string> = {};
+const getNextNodeTemplateId = () => ++id;
+
+@inject(Element, Container, StyleEngine, UxDefaultTreeViewConfiguration)
+@customElement('ux-tree-view')
+@useView(PLATFORM.moduleName('./ux-tree-view.html'))
+@processContent(UxTreeView.processContent)
+export class UxTreeView implements UxComponent {
+  static processContent(_viewCompiler: ViewCompiler, _resources: ViewResources, element: Element, _instruction: BehaviorInstruction) {
+    const treeNode = element.querySelector('ux-tree-node');
+    if (treeNode) {
```

Comment:
I'm wondering if we should have this logic:
- if there is a `<tree-node/>`, then we use the content of that as template
- else, we use all the children of `<tree-view/>` as the content of the inner `tree-node` template
Probably it can help reduce the boilerplate of having to write `<tree-node/>`, though it would be really ambiguous as you don't know what the content is for

Reply:
I considered this. I think it's better to be explicit here

Reply:
Yeah, maybe it's best to do that here.